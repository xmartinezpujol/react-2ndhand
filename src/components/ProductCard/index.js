import React from 'react';
import PropTypes from 'prop-types';

import { Link } from 'react-router-dom';

import { Card, Icon } from 'semantic-ui-react';

import {
  CardDescription,
  CardInfo,
  CardWrapper,
  LabelAdapter,
  Title,
} from './styles';

const ProductCard = props => (
  <CardWrapper link>
    <Card.Content as={Link} href={`/list/${props.data.id}`} to={`/list/${props.data.id}`}>
      <Card.Header>
        <Title>
          {props.data.title}
        </Title>
      </Card.Header>
      <Card.Meta>
        {props.data.params.map(param => (
          <LabelAdapter key={`${props.data.id}-${param[0]}`} color="teal">{param[0]}
            <LabelAdapter.Detail>{param[1]}</LabelAdapter.Detail>
          </LabelAdapter>
        ))}
      </Card.Meta>
      <CardDescription>{props.data.description.length > 100
        ? `${props.data.description.slice(0, 300)}...`
        : props.data.description}
      </CardDescription>
    </Card.Content>
    <Card.Content extra>
      <Icon color="teal" name="time" />
      <CardInfo>{`${props.data.created}`}</CardInfo>
      <Icon color="teal" name="map marker" />
      <CardInfo>{`${props.data.city_label}`}</CardInfo>
    </Card.Content>
  </CardWrapper>
);

ProductCard.propTypes = {
  data: PropTypes.shape({
    city_label: PropTypes.string,
    created: PropTypes.string,
    description: PropTypes.string,
    id: PropTypes.string,
    title: PropTypes.string,
    params: PropTypes.array,
  }).isRequired,
};

export default ProductCard;
