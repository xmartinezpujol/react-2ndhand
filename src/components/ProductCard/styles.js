import styled from 'styled-components';

import { Card, Label } from 'semantic-ui-react';

export const CardWrapper = styled(Card)`
  &&& {
    height: 280px;
    width: 100%;
    margin: 0;
    margin-bottom: 20px;
    @media (min-width: 768px) {
      width: calc(50% - 20px);
    }
    
    @media (min-width: 992px) {
      width: calc(33% - 15px);
    }
  }
`;

export const CardDescription = styled(Card.Description)`
  &&& {
    line-clamp: 3;
  }
`;

export const CardInfo = styled.span`
  margin-right: 10px;
`;

export const LabelAdapter = styled(Label)`
  display: flex;
  font-size: 10px !important;
`;

export const Title = styled.div`
  font-size: 16px;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  margin-bottom: 10px;
`;
