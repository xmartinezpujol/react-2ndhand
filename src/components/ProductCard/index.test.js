import React from 'react';
import { mount } from 'enzyme';
import { StaticRouter } from 'react-router';
import renderer from 'react-test-renderer';

// $FlowFixMe
import Ads from '../../mocks/ads.json';

import ProductCard from './index';

describe('ProductCard', () => {
  const addsByPage = Ads.map(page => page.ads);
  const product = [].concat.apply([], addsByPage)[0];

  it('renders correctly', () => {
    const context = {};
    const tree = renderer.create(
      <StaticRouter location="/list" context={context}>
        <ProductCard product={product} />
      </StaticRouter>).toJSON();
    const wrapper = mount(
      <StaticRouter location="/list" context={context}>
        <ProductCard product={product} />
      </StaticRouter>);
    expect(tree).toMatchSnapshot();
    expect(wrapper.find('CardHeader').length).toBe(1);
    expect(wrapper.find('Label').length).toBe(2);
    expect(wrapper.find('CardMeta').length).toBe(1);
    expect(wrapper.find('CardDescription').length).toBe(1);
    expect(wrapper.find('Icon').length).toBe(2);
    expect(wrapper.find('CardDescription').getElement().props.children.length).toBe(303);
  });
});

