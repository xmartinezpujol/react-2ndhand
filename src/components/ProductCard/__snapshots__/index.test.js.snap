// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard renders correctly 1`] = `
<div
  className="ui link card styles__CardWrapper-sc-13wxosj-0 ksZAYB"
  data-cy="productCard"
  onClick={[Function]}
>
  <a
    className="content"
    href="/list/205799123"
    onClick={[Function]}
  >
    <div
      className="header"
    >
      <div
        className="styles__Title-sc-13wxosj-4 gXnukw"
      >
        iphone 5s 16gb silver bialy stan dbd Pl dystrybucja 100% sprawny
      </div>
    </div>
    <div
      className="meta"
    >
      <div
        className="ui teal label styles__LabelAdapter-sc-13wxosj-3 drDSqv"
        onClick={[Function]}
      >
        Marka
        <div
          className="detail"
        >
          iPhone
        </div>
      </div>
      <div
        className="ui teal label styles__LabelAdapter-sc-13wxosj-3 drDSqv"
        onClick={[Function]}
      >
        Stan
        <div
          className="detail"
        >
          Używane
        </div>
      </div>
    </div>
    <div
      className="description styles__CardDescription-sc-13wxosj-1 jJIAfJ"
    >
      Sprzedam Iphone 5s w bdb stanie technicznym i wizualnym (jak na foto) telefon jest w 100 % sprawny wszystko dziala nie zawiesza sie i nie gubi sieci. Wizualnie jak na zdjeciach posiada normalne slady urzytkowania,
Pl dystrybucja telefon fabrycznie bez blokady simlock. 
Telefon sprzedaje w kmpl 
W sk...
    </div>
  </a>
  <div
    className="extra content"
  >
    <i
      aria-hidden="true"
      className="teal time icon"
      onClick={[Function]}
    />
    <span
      className="styles__CardInfo-sc-13wxosj-2 hDerPd"
    >
      10  lut
    </span>
    <i
      aria-hidden="true"
      className="teal map marker icon"
      onClick={[Function]}
    />
    <span
      className="styles__CardInfo-sc-13wxosj-2 hDerPd"
    >
      Kobyłka
    </span>
  </div>
</div>
`;
